lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1763,14 +1763,18 @@ def _shade_colors(self, color, normals, lightsource=None):
         mask = ~np.isnan(shade)
 
         if len(shade[mask]) > 0:
-            norm = Normalize(min(shade[mask]), max(shade[mask]))
-            shade[~mask] = min(shade[mask])
+            # convert dot product to allowed shading fractions
+            in_norm = Normalize(-1, 1)
+            out_norm = Normalize(0.3, 1).inverse
+            norm = lambda x: out_norm(in_norm(x))
```

Comment:
PEP 8 doesn't like assigning a variable to a lambda. I'm kind of indifferent but I guess it would be possible to do
```python
def norm(x):
    return out_norm(in_norm(x))
```

Reply:
Has the extra requirement that it needs blank lines around it, which IMO makes the grouping less clear.

I guess I'll just drop the `lambda`, and substitute `out_norm(in_norm(x))` at the call site.